remotespark/livyclientlib/sqlquery.py

```diff
@@ -0,0 +1,40 @@
+import remotespark.utils.configuration as conf
+
+
+class SQLQuery(object):
+    def __init__(self, query, samplemethod=None, maxrows=None,
+                 samplefraction=None, only_columns=False):
+        if samplemethod is None:
+            samplemethod = conf.default_samplemethod()
+        if maxrows is None:
+            maxrows = conf.default_maxrows()
+        if samplefraction is None:
+            samplefraction = conf.default_samplefraction()
+
+        assert samplemethod == 'take' or samplemethod == 'sample'
+        assert isinstance(maxrows, int)
```

Comment:
Yes, if maxrows is negative that means that the number of rows pulled from the server is not limited.